mobile/native/src/channel_fee.rs

```diff
@@ -0,0 +1,169 @@
+use crate::commons::reqwest_client;
+use crate::config;
+use crate::event::subscriber::Subscriber;
+use crate::event::EventInternal;
+use crate::event::EventType;
+use crate::ln_dlc;
+use anyhow::anyhow;
+use anyhow::Result;
+use bitcoin::Txid;
+use ln_dlc_node::node::rust_dlc_manager::ChannelId;
+use serde::Deserialize;
+use serde::Serialize;
+use std::sync::Arc;
+use std::sync::Mutex;
+use tokio::runtime::Handle;
+
+#[derive(Clone)]
+pub struct ChannelFeePaymentSubscriber {
+    pub open_channel_tx: Arc<Mutex<Option<EsploraTransaction>>>,
```

Comment:
🤔 I'm surprised that we need an `Arc<Mutex<_>>` here.

Reply:
🔧 Perhaps relatedly, we could model this as an enum:

```rust
enum ChannelOpeningFeeState {
    PendingOpen, // before `ChannelReady`
    Owing { fee: u64 }, // after `ChannelReady`, before first `PaymentClaimed`
    Paid, // after first `PaymentClaimed`
}
```

It might help make the code clearer. For instance, every time we get a new `PaymentClaimed` and we have already `Paid` we can be sure that we can safely ignore the event. But if we get it at `PendingOpen` we can emit an error as something has gone very wrong.

Reply:
I like the idea, but given that this is only an intermediary solution, I am not going to spent more effort into improving the design.